server/lib/pbench/process_tb.py

```diff
@@ -0,0 +1,113 @@
+import os
+import subprocess
+
+from collections import namedtuple
+from logging import Logger
+from pathlib import Path
+
+from pbench import PbenchConfig
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+class ProcessTb:
+    def __init__(self, config: PbenchConfig, logger: Logger):
+        """ Processes Tar ball, gets the tar ball from receiving directory
+        and copies it to the remote server by creating a subprocess which
+        call Agent's 'pbench-results-push' command.
+
+            Args -
+                config -- PbenchServer config object
+                logger -- logger object to use when emitting log entries during
+                            operation
+
+        """
+        self.config = config
+        self.logger = logger
+        # receive_dir -- directory where tarballs are received from agent
+        self.receive_dir = self.get_receive_dir()
+
+    def get_receive_dir(self) -> Path:
+        receive_dir_prefix = self.config.get(
+            "pbench-server", "pbench-receive-dir-prefix"
+        )
+        if not receive_dir_prefix:
+            raise ValueError(
+                "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+            )
+
+        receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+        if not receive_dir.is_dir():
+            raise NotADirectoryError(
+                "Failed: {} does not exist, or is not a directory", receive_dir
+            )
+
+        return receive_dir
+
+    def results_push(self, controller: str, tb: Path, token: str):
```

Comment:
This is also a helper function, and so can be prefixed by an underscore to indicate it as such.

Reply:
It also does not have to be a class method since it is just a simple wrapper around `subprocess.run()` for convenient error handling.

Reply:
It needs to be patchable so that it can be mocked (otherwise, there's no need to have it in a separate function at all).

It could be a static (class) method instead of an instance method, but I think it will make things tricky if we change it from a method to a simple function.

Reply:
> It could be a static (class) method instead of an instance method, but I think it will make things tricky if we change it from a method to a simple function.

I agree with Webb.

Reply:
Peter is right.  Riya, you were able to mock `os.remove()` successfully -- I'm fairly sure that that is just a plain function in a package.  So, as long as you can reference the function in its defining scope, you should be able to mock it.

That aside, since this function is only used within this class, I like having it as a method; since it doesn't actually reference the `self` parameter, we could make it be a `@static`/class method instead of an instance method...which, I think, is actually preferable.  That does mean that, in the unit test context, when you insert a mock for it, that mock affects all instances of the class, but that's fine (even good).

Reply:
Renaming this to `_results_push` and making it a static method would be sufficient.